### ISSUE TITLE
Boost-regex[icu]

### ISSUE THREAD

**Issue by jm130794** [microsoft/vcpkg#4962]
Hello,

I can't install boost-regex[icu] on Ubuntu 16.04.

```
~/vcpkg$ vcpkg update
Using local portfile versions. To update the local portfiles, use `git pull`.
No packages need updating.
jm@noisette:~/vcpkg$ git pull
Already up-to-date.
jm@noisette:~/vcpkg$ vcpkg upgrade
All installed packages are up-to-date with the local portfiles.
jm@noisette:~/vcpkg$ vcpkg install boost-regex[icu]
The following packages will be built and installed:
    boost-regex[core,icu]:x64-linux
Starting package 1/1: boost-regex:x64-linux
Building package boost-regex[core,icu]:x64-linux... 
-- Using cached /home/jm/vcpkg/downloads/boostorg-regex-boost-1.68.0.tar.gz
-- Using source at /home/jm/vcpkg/buildtrees/boost-regex/src/ost-1.68.0-e671fb14f8
-- Configuring x64-linux-dbg
-- Configuring x64-linux-rel
-- Building x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_build_cmake.cmake:175 (message):
    Command failed: /donnees/jm/vcpkg/downloads/tools/cmake-3.12.4-linux/cmake-3.12.4-Linux-x86_64/bin/cmake;--build;.;--config;Debug;--target;install;--;-v;-j1
    Working Directory: /home/jm/vcpkg/buildtrees/boost-regex/x64-linux-dbg
    See logs for more information:
      /home/jm/vcpkg/buildtrees/boost-regex/install-x64-linux-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
  installed/x64-linux/share/boost-build/boost-modular-build.cmake:82 (vcpkg_install_cmake)
  ports/boost-regex/portfile.cmake:20 (boost_modular_build)
  scripts/ports.cmake:71 (include)


Error: Building package boost-regex:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: boost-regex:x64-linux
  Vcpkg version: 2018.11.23-unknownhash

Additionally, attach any relevant sections from the log files above.
jm@noisette:~/vcpkg$
```

Jean-Marc

**Comment by myd7349** [comment]
Could you attach the log file mentioned in the error message?

**Comment by luzpaz** [comment]
There is a pending PR (#5059) does this issue persist there ?

**Comment by JackBoosY** [comment]
Error info:
```
error: Unable to find file or target named
error:     '/debug/lib/icudtd.lib'
error: referred to from project at
error:     ''
```

It's boost-build issue.

**Comment by JackBoosY** [comment]
In _ports/boost-modular-build-helper/boost-modular-build.cmake_(Line261-262):
```
    configure_file(${_bm_DIR}/user-config.jam ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg/user-config.jam @ONLY)
    configure_file(${_bm_DIR}/user-config.jam ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/user-config.jam @ONLY)
```

In _ports/boost-modular-build-helper/user-config.jam_(Line42-52):
```
lib icuuc : : <file>"@CURRENT_INSTALLED_DIR@/lib/icuuc.lib" <variant>release : : <library>advapi32 ;
lib icuuc : : <file>"@CURRENT_INSTALLED_DIR@/debug/lib/icuucd.lib" <variant>debug : : <library>advapi32 ;

lib icuin : : <file>"@CURRENT_INSTALLED_DIR@/lib/icuin.lib" <variant>release : : ;
lib icuin : : <file>"@CURRENT_INSTALLED_DIR@/debug/lib/icuind.lib" <variant>debug : : ;

lib icudt : : <file>"@CURRENT_INSTALLED_DIR@/lib/icudt.lib" <variant>release : : ;
lib icudt : : <file>"@CURRENT_INSTALLED_DIR@/debug/lib/icudtd.lib" <variant>debug : : ;

lib iconv : : <file>"@LIBICONV_LIBRARY@" <link>shared <runtime-link>shared <variant>release : : ;
lib iconv : : <file>"@LIBICONV_LIBRARY@" <link>shared <runtime-link>shared <variant>debug : : ;
```

Error:
1. The icu libraries name are incorrect in Linux.
2. CURRENT_INSTALLED_DIR configuration failed.

**Comment by strega-nil** [comment]
Can we see if this works now @jm130794 ? I think that #9780 should have fixed it maybe?

**Comment by JackBoosY** [comment]
Hi @jm130794, thanks for reporting this issue.
This issue should fixed in #9780, please update vcpkg and rebuild boost-regex[icu].

Thanks.
